File: Day_17/Q66.java
import java.util.Scanner;

class Q66{

    static Scanner sc = new Scanner(System.in);

    static void input_array(int arr[],int n){
        for(int i = 0;i < n;i++){
            arr[i] = sc.nextInt();
        }
    }

    static void display_array(int arr[]){
        for(int i = 0;i < arr.length;i++){
            System.out.print(arr[i] + "\t");
        }
        System.out.println();
    }

    //Check Element Present
    static boolean is_present(int arr[],int size,int target){

        for(int i = 0;i < size;i++){
            if(arr[i] == target){
                return true;
            }
        }

        return false;
    }

    //Union Of Arrays
    static int[] union_array(int arr1[],int arr2[]){

        int temp[] = new int[arr1.length + arr2.length];

        int index = 0;

        //Insert First Array
        for(int i = 0;i < arr1.length;i++){
            temp[index++] = arr1[i];
        }

        //Insert Unique Elements
        for(int i = 0;i < arr2.length;i++){

            if(is_present(temp,index,arr2[i]) == false){
                temp[index++] = arr2[i];
            }
        }

        int uni[] = new int[index];

        for(int i = 0;i < index;i++){
            uni[i] = temp[i];
        }

        return uni;
    }

    public static void main(String args[]){

        System.out.print("Enter Size Of First Array : ");
        int n1 = sc.nextInt();

        int arr1[] = new int[n1];

        System.out.print("Enter Elements Of First Array : ");
        input_array(arr1,n1);

        System.out.print("Enter Size Of Second Array : ");
        int n2 = sc.nextInt();

        int arr2[] = new int[n2];

        System.out.print("Enter Elements Of Second Array : ");
        input_array(arr2,n2);

        int uni[] = union_array(arr1,arr2);

        System.out.print("Union Array : ");
        display_array(uni);
    }
    sc.close();
}

File: Day_17/Q66.cpp
#include <bits/stdc++.h>
using namespace std;

void input_array(vector<int> &arr,int n){
    for(int i = 0;i < n;i++) cin >> arr[i];
}

void display_array(vector<int> &arr){
    for(int i = 0;i < arr.size();i++){
        cout << arr[i] << "\t";
    }
    cout << endl;
}

//Union Of Arrays
vector<int> union_array(vector<int> arr1,vector<int> arr2){
    vector<int> uni;

    //Insert First Array
    for(int i = 0;i < arr1.size();i++){
        uni.push_back(arr1[i]);
    }

    //Insert Unique Elements Of Second Array
    for(int i = 0;i < arr2.size();i++){

        bool found = false;

        for(int j = 0;j < uni.size();j++){
            if(arr2[i] == uni[j]){
                found = true;
                break;
            }
        }

        if(found == false){
            uni.push_back(arr2[i]);
        }
    }

    return uni;
}

int main(){

    int n1,n2;

    cout << "Enter Size Of First Array : ";
    cin >> n1;

    vector<int> arr1(n1);

    cout << "Enter Elements Of First Array : ";
    input_array(arr1,n1);

    cout << "Enter Size Of Second Array : ";
    cin >> n2;

    vector<int> arr2(n2);

    cout << "Enter Elements Of Second Array : ";
    input_array(arr2,n2);

    vector<int> uni = union_array(arr1,arr2);

    cout << "Union Array : ";
    display_array(uni);

    return 0;
}
